File: frontend/src/app/features/auth/password-policy.ts
// Mirrors `PasswordPolicy` on the backend, which is where the rule actually
// lives — these are the *message*, not the guarantee. `POST /auth/activate`
// validates the same bounds and answers 400 with a per-field error regardless
// of what this file says.
//
// Worth mirroring at all because the alternative is telling somebody their
// password is too short only after a round trip, on the one screen where they
// are inventing a password rather than recalling one.
//
// 12 to 128, length only, no composition rules. That follows NIST SP 800-63B,
// and the maximum is not cosmetic: the server runs PBKDF2 at 100k iterations
// over whatever it is given, so an unbounded password is a cheap way to make an
// anonymous endpoint do unbounded work. The reasoning is recorded in full
// beside the server-side constants; it was invented in user management phase 2
// because nothing in the PRD specifies one.
export const PASSWORD_MIN_LENGTH = 12;
export const PASSWORD_MAX_LENGTH = 128;

File: backend/src/BookSpace.Application/Features/Authentication/PasswordPolicy.cs
namespace BookSpace.Application.Features.Authentication;

// What this application will accept as a password.
//
// **This is the first place BookSpace ever sets one**, so the policy had to be
// invented here — FR-2.3 says passwords are hashed and never stored in
// plaintext, and neither it nor anything else in the PRD says how long or how
// complex one must be. Recorded as a deliberate choice rather than left as
// magic numbers in a validator, and flagged to the owner rather than presented
// as a requirement (docs/user-management-plan.md §5, phase 2).
//
// Length only, no composition rules — no "must contain an uppercase letter and
// a digit". That follows NIST SP 800-63B, which dropped composition rules
// because they push people towards predictable substitutions (Password1!) while
// doing little for entropy; length is what actually helps. 12 is the shortest
// that is defensible today for a credential with no second factor.
//
// The maximum is not cosmetic. IPasswordHasher runs PBKDF2 at 100k iterations
// over whatever it is given, so an unbounded password is a cheap way to make
// the server do unbounded work on an anonymous endpoint.
//
// One place, because the rule has to be the same everywhere a password is set.
// Today that is activation; a password reset or a self-service change (both out
// of scope, §3.4) would be the next callers, and a second copy is how they
// would come to disagree.
public static class PasswordPolicy
{
    public const int MinimumLength = 12;

    public const int MaximumLength = 128;
}
